src/PoolManager.sol

```diff
@@ -207,34 +218,67 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
         PoolId id = key.toId();
         _checkPoolInitialized(id);
 
-        key.hooks.beforeSwap(key, params, hookData);
-
-        uint256 feeForProtocol;
-        uint24 swapFee;
-        Pool.SwapState memory state;
-        (delta, feeForProtocol, swapFee, state) = pools[id].swap(
-            Pool.SwapParams({
-                tickSpacing: key.tickSpacing,
-                zeroForOne: params.zeroForOne,
-                amountSpecified: params.amountSpecified,
-                sqrtPriceLimitX96: params.sqrtPriceLimitX96
-            })
-        );
-
-        _accountPoolBalanceDelta(key, delta);
-
-        // the fee is on the input currency
-        unchecked {
-            if (feeForProtocol > 0) {
-                protocolFeesAccrued[params.zeroForOne ? key.currency0 : key.currency1] += feeForProtocol;
+        // The hook's deltas are from the point of view of the hook. Positive: the hook took money, negative: the hook sent money to the pool
+        // TODO consider adding hookDeltaInUnspecified to beforeSwap hook too
+        int128 hookDeltaInSpecified = key.hooks.beforeSwap(key, params, hookData);
+
+        bool exactInput = params.amountSpecified < 0;
```

Comment:
Based on the comment below isn't this suppose to be the other way?
```
// ExactIn (amountSpecified is positive)
```

```suggestion
        bool exactInput = params.amountSpecified > 0;
```

Reply:
the signs are being flipped! generally, a negative sign will indicate user's token balance going down

looks like the code comment will be stale

https://github.com/Uniswap/v4-core/pull/481